src/signTx.c

```diff
@@ -22,6 +22,40 @@ enum {
 };
 
 static ins_sign_tx_context_t* ctx = &(instructionState.signTxContext);
+static ins_sign_tx_body_context_t* txBodyCtx = &(instructionState.signTxContext.txPartCtx.body_ctx);
+static ins_sign_tx_aux_data_context_t* txAuxDataCtx = &(instructionState.signTxContext.txPartCtx.aux_data_ctx);
+static ins_sign_tx_witnesses_context_t* txWitnessesCtx = &(instructionState.signTxContext.txPartCtx.witnesses_ctx);
+
+// TODO - maybe add an enum to the global context which would specify the active tx part?
```

Comment:
I'd appreciate your insight on this @janmazak

Reply:
as discussed offline, controlling access to the corresponding tx part based on an enum and wrapping accessor function seems worthwile, but it would bloat this PR with too many changes, so we'd do it in another PR - I made an issue for it: https://github.com/vacuumlabs/ledger-app-cardano-shelley/issues/66